### PR TITLE
Add option to specify environment type

### PR DESCRIPTION
**Pull request by skolsuper**
There are two types of beanstalk, LoadBalanced and SingleInstance.  Not entirely sure if the SingleInstance type is ever a great choice, but I use it sometimes for when I want the ElasticBeanstalk infrastructure sugar without the horizontal scaling, e.g. right now for running a slackbot

### CODE REVIEW

**Comment by skolsuper** [comment]
I got around to using this in production and found some extra caveats I added to the documentation.  Maybe if you have a major version coming up, you can consider renaming `private_subnets` and `public_subnets` to `instance_subnets` (or just `subnets`) and `elb_subnets` respectively, to be more explicit about their purpose.

File: main.tf [hunk 1]
```diff
@@ -440,39 +440,39 @@ resource "aws_elastic_beanstalk_environment" "default" {
   ###=========================== Autoscale trigger ========================== ###
 
   setting {
-    namespace = "aws:autoscaling:trigger"
-    name      = "MeasureName"
-    value     = "${var.autoscale_measure_name}"
+    namespace = "${var.environment_type == "LoadBalanced" ? "aws:autoscaling:trigger" : ""}"
```
Comment (aknysh) [inline main.tf]: does it work when with all those `namespace`, `name`, `value` set to empty strings?
(when var.environment_type == "SingleInstance")
TF does not complain?

Comment (aknysh) [reply main.tf]: @skolsuper  since you on it, can you please add `description` to `wait_for_ready_timeout` variable and rebuild README again?
thanks a lot

Comment (skolsuper) [reply main.tf]: Yep :-). I got here through much trial and error...

**Comment by aknysh** [comment]
to update README, no need to install `gomplate` manually, just run these three commands:

```
make init
make readme/deps
make readme
```

**Comment by skolsuper** [comment]
Given #43, its planned solution once Terraform 0.12 arrives, and the number of possible permutations of loadbalancer types: should I revert all the ugly ternary conditionals?

**Pull request merged**